src/component_macro_reference.md

```diff
@@ -0,0 +1,228 @@
+
+```ignore
+set_property_name[value1, value2, ...] = gtk::Box { ... }
+```
+
+This will expand to `parent.set_property_name(box_widget, value1, value2, ...)`.
+
+### Naming widgets
+
+Widgets may be given a name with the `name` attribute. These names are accessible as fields on the `Widgets` struct generated by the macro.
+
+```ignore
+#[name = "important_label"]
+gtk::Label { ... }
+
+// ...
+
+let widgets = view_output!();
+let label = &widgets.important_label;
+```
+
+Names can also be assigned with this syntax:
+
+```ignore
+set_child: important_label = gtk::Label { ... }
+```
+
+### Properties
+
+Properties are initialized and mutated by calling methods within the widget types. See the documentation for each widget type to see what methods are available. Generally properties are set via setter methods, but any methods on the widget can be called, too. Many of these methods are part of an extension trait associated with the widget type. Those traits must be in scope to call their methods. For example, if you want to use the [`set_default_width`](https://gtk-rs.org/gtk4-rs/git/docs/gtk4/prelude/trait.GtkWindowExt.html#tymethod.set_default_width) method from the `GtkWindowExt` trait, you must import the trait directly or glob import it from the prelude (`use gtk::prelude::*;`).
+
+To initialize a property with a value:
+
+```ignore
+set_property_name: value,
+```
+
+Initialize a property only if its value is `Some`, and do nothing if it's `None`:
+
+```ignore
+set_property_name?: value,
+```
+
+Call a method that has multiple arguments:
+
+```ignore
+set_property_name: (value1, value2, ...),
+```
+
+Initialize and automatically update a property.
+
+```ignore
+#[watch]
+set_property_name: (value1, value2, ...),
+```
+
+Initialize and automatically update a property with a tracker. The `track_expression` can be any expression that returns a `bool`. If it's true, it indicates that the method should be called:
+
+```ignore
+#[track(track_expression)]
+set_property_name: (value1, value2, ...),
+```
+
+Initialize a property by iterating over an iterator. You can use this for repeated calls to setter functions, like `add_class_name` in case you have multiple class names in a `Vec`.
+
+```ignore
+#[iterate]
+set_property_name: iterator,
+```
+
+#### Trait disambiguation
+
+Sometimes, several traits implement the same method for a type. In this case, you need to tell Rust which trait it should use. For example, the `set_child` function is implemented by both `gtk::prelude::GtkWindowExt` and `libadwaita::traits::ApplicationWindowExt`. If we use the regular syntax, the Rust compiler will get confused and tells us to specify the trait. To specify the intended trait, we use the `TraitName::method` syntax that's similar to [Rust's fully qualified syntax for trait disambiguation](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name).
```

Comment:
The example is outdated meanwhile and should be removed or replaced. The libadwaita devs were kind enough to rename the `set_child` method to `set_content` to avoid naming conflicts.

Reply:
I see. Removed the specific example, but left the rest of the text.